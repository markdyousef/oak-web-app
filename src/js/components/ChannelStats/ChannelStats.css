:root {
    --main-black-color: #000;
    --main-white-color: #fefeff;
    --main-orange-color: #ff9200;
    --main-red-color: #8b0000;
    --main-green-color: #005d6c;
}

.container {
    width: 350px;
    height: 250px;
    margin: 10px;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
    background-color: #fff;
}

.container:hover {
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
}

.container header {
    width: 100%;
}

header h1 {
    padding: 15px;
    font-size: 14px;
    color: var(--main-green-color);
}

.purpose {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.purpose h2 {
    font-size: 12px;
    font-weight: lighter;
    width: 90%;
}

.purpose h4 {
    padding: 14px;
    color: var(--main-red-color);
    font-size: 12px;
    align-self: flex-end;
}

.statsRow {
    width: 100%;
    display: flex;
    justify-content: space-around;
}

.statsRow div {
    width: 100px;
    margin: 15px;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.statsRow img {
    height: 30px;
}
