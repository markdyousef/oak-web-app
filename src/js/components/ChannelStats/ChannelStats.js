// @flow
import React, { PropTypes } from 'react';
import { Link } from 'react-router';

import css from './ChannelStats.css';
import chatIcon from '../../../img/chat.png';
import membersIcon from '../../../img/multiple-users-silhouette.png';

const ChannelStats = ({ data } : Object) => {
    const { name, purpose, members, messages, id } = data;

    return (
        <Link className={css.container} to={`/channel/${id}`}>
            <header>
                <h1>{name}</h1>
                {purpose && purpose.value &&
                    <div className={css.purpose}>
                        <h2>{`"${purpose.value}"`}</h2>
                        <h4>{purpose.creator}</h4>
                    </div>}
            </header>
            <div className={css.statsRow}>
                {members &&
                    <div>
                        <img src={membersIcon} alt="members" />
                        <h5>{members.length}</h5>
                    </div>
                }
                {messages &&
                    <div>
                        <img src={chatIcon} alt="chat" />
                        <h5>{messages.length}</h5>
                    </div>
                }
            </div>
        </Link>
    );
};

ChannelStats.propTypes = {
    data: PropTypes.shape({
        name: PropTypes.string.isRequired,
        purpose: PropTypes.shape({
            value: PropTypes.string.isRequired,
            creator: PropTypes.string.isRequired
        }),
        members: PropTypes.arrayOf(PropTypes.string),
        messages: PropTypes.arrayOf(PropTypes.object)
    }).isRequired
};

export default ChannelStats;
