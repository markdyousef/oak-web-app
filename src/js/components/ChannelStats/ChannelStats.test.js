// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import ChannelStats from './ChannelStats';

test('renders ChannelStats', () => {
    const props = {
        data: {
            name: 'general'
        }
    };
    const component = renderer.create(
        <ChannelStats {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
