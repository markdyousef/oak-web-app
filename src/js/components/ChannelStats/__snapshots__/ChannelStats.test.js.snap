exports[`test renders ChannelStats 1`] = `
<div
  className={undefined}>
  <header>
    <h1>
      general
    </h1>
  </header>
  <div
    className={undefined} />
</div>
`;
